src/core/shape_inference/include/eye_shape_inference.hpp

```diff
@@ -66,7 +67,8 @@ std::vector<TRShape> shape_infer(const Eye* op,
                 output_shape = PartialShape::dynamic(batch_shape[0].get_length());
             }
         } else {
-            return {ov::PartialShape::dynamic()};
+            output_shape = PartialShape::dynamic();
```

Comment:
It's redundant to `std::vector` ctor job. Why is `PartialShape` used in here anyway?

Reply:
PartialShape beause it dynamic case.
It only replace single element in vector not create new one.